packages/components/src/color-picker/stories/index.tsx

```diff
@@ -1,14 +1,17 @@
+/**
+ * External dependencies
+ */
+import type { ComponentMeta, ComponentStory } from '@storybook/react';
 /**
  * WordPress dependencies
  */
 import { useState } from '@wordpress/element';
-
 /**
  * Internal dependencies
  */
 import { ColorPicker } from '..';
 
-export default {
+const meta: ComponentMeta< typeof ColorPicker > = {
```

Comment:
We should leverage TS for generating docs, but at the same time, we should hide those legacy props — something like this may work

```diff
diff --git a/packages/components/src/color-picker/stories/index.tsx b/packages/components/src/color-picker/stories/index.tsx
index 00ce542360..187124c591 100644
--- a/packages/components/src/color-picker/stories/index.tsx
+++ b/packages/components/src/color-picker/stories/index.tsx
@@ -20,6 +20,21 @@ const meta: ComponentMeta< typeof ColorPicker > = {
 			control: { type: 'select' },
 			options: [ 'rgb', 'hsl', 'hex', undefined ],
 		},
+		onChangeComplete: {
+			table: {
+				disable: true,
+			},
+		},
+		oldHue: {
+			table: {
+				disable: true,
+			},
+		},
+		disableAlpha: {
+			table: {
+				disable: true,
+			},
+		},
 		// We can't use a `on*` regex because this component will switch to
 		// legacy mode when an onChangeComplete prop is passed.
 		onChange: { action: 'onChange' },

```

Reply:
✅  feba5ddcb2

Reply:
Actually, it still looks like we could do better — what if, instead of using the legacy adapter, we used the new picker in Storybook? That way, we'd get automatically the docs for just the new props, and we would be able to delete the "hacky" configs that I suggested above.

<details>

<summary>Something like this maybe?</summary>

```diff
diff --git a/packages/components/src/color-picker/component.tsx b/packages/components/src/color-picker/component.tsx
index 379b8557d0..852e0756b3 100644
--- a/packages/components/src/color-picker/component.tsx
+++ b/packages/components/src/color-picker/component.tsx
@@ -38,7 +38,7 @@ const options = [
 	{ label: 'Hex', value: 'hex' as const },
 ];
 
-const ColorPicker = (
+const UnconnectedColorPicker = (
 	props: ColorPickerProps,
 	forwardedRef: ForwardedRef< any >
 ) => {
@@ -112,6 +112,9 @@ const ColorPicker = (
 	);
 };
 
-const ConnectedColorPicker = contextConnect( ColorPicker, 'ColorPicker' );
+export const ColorPicker = contextConnect(
+	UnconnectedColorPicker,
+	'ColorPicker'
+);
 
-export default ConnectedColorPicker;
+export default ColorPicker;
diff --git a/packages/components/src/color-picker/stories/index.tsx b/packages/components/src/color-picker/stories/index.tsx
index bb4adb5410..419c719571 100644
--- a/packages/components/src/color-picker/stories/index.tsx
+++ b/packages/components/src/color-picker/stories/index.tsx
@@ -2,44 +2,26 @@
  * External dependencies
  */
 import type { ComponentMeta, ComponentStory } from '@storybook/react';
+
 /**
  * WordPress dependencies
  */
 import { useState } from '@wordpress/element';
+
 /**
  * Internal dependencies
  */
-import { ColorPicker } from '..';
+import { ColorPicker } from '../component';
 
 const meta: ComponentMeta< typeof ColorPicker > = {
 	component: ColorPicker,
 	title: 'Components/ColorPicker',
 	argTypes: {
+		as: { control: { type: null } },
 		color: { control: { type: null } },
-		copyFormat: {
-			control: { type: 'select' },
-			options: [ 'rgb', 'hsl', 'hex', undefined ],
-		},
-		onChangeComplete: {
-			table: {
-				disable: true,
-			},
-		},
-		oldHue: {
-			table: {
-				disable: true,
-			},
-		},
-		disableAlpha: {
-			table: {
-				disable: true,
-			},
-		},
-		// We can't use a `on*` regex because this component will switch to
-		// legacy mode when an onChangeComplete prop is passed.
-		onChange: { action: 'onChange' },
 	},
 	parameters: {
+		actions: { argTypesRegex: '^on.*' },
 		controls: {
 			expanded: true,
 		},

```
</details>